### ISSUE TITLE
promptUserForID not defined?

### ISSUE THREAD

**Issue by warrenalphonso** [t-mullen/simple-signal#16]
I'm fairly new to simple-peer and node as a whole. Am I supposed to define this myself?

**Comment by t-mullen** [comment]
Discovery and signalling data sent through simple-signal is over the server, but data written to peer objects is sent P2P. The simple-peer objects are the P2P part of this library.

If you want a full mesh (all peers connected to each other), you can make that with a small change to the demo code. Just connect to every ID instead of picking one:

```javascript
signalClient.on('discover', async (allIDs) => {
  allIDS.forEach(id => {
    const { peer } = await signalClient.connect(id) // connect to target client
    peer // this is a fully-signaled simple-peer object (initiator side)
  })
})
```

**Comment by warrenalphonso** [comment]
Is there a way to send data to peer objects (like a video stream) through the server instead of P2P? If not, is there any faster way to do this aside from full-mesh?

**Comment by t-mullen** [comment]
You can forward data through server using socket.io. It's just ordinary websockets then.

There's lots of other topologies and algorithms that aren't fully-connect mesh. They have different tradeoffs. Full mesh makes the most connections but has the lowest latency. What kind of application are you trying to make?

**Comment by warrenalphonso** [comment]
A group video chat. I wasn't aware socket.io takes video streams and can't find a lot of info on that. Could you link me to some resources if possible?

**Comment by t-mullen** [comment]
socket.io doesn't take media streams, I thought you meant data streams. You'll need a media server like Kurento: http://www.kurento.org/

P2P full-mesh (with no media server) is probably the easiest way to get a video chat working though. You should be able to get ~5 people talking reliably.

**Comment by warrenalphonso** [comment]
Okay thanks so much for being willing to explain. This really helped me out!

**Comment by warrenalphonso** [comment]
I'm trying to figure out a way to clean up a peer connection when a peer just closes their browser or some other error occurs. simple-peer's `peer.on('error', function(err){})` works but it takes a long time to execute even for just 2 peers communicating. I'm not sure if I've messed something up here or if it normally takes that long, but I noticed simple-signal-server detects a disconnected user really fast with the disconnect listener. Is there a way I can somehow access peer.destroy() in the signalServer disconnect listener?
